scripts/2.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,535 @@
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[14]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'samples.tsv')\nclinical = pd.read_table(path, index_col=0)")
+
+
+# In[15]:
+
+clinical.tail(5)
+
+
+# In[16]:
+
+# Subset the Y matrix to only the genes to be classified
+y_full = Y[classifier_genes_df['entrez_gene_id']]
+
+
+# In[17]:
+
+y_full.columns = classifier_genes_df['gene_symbol']
+y_full['disease'] = clinical['disease']
+
+
+# In[18]:
+
+mut_tis_total = y_full.groupby('disease').sum()
+mut_tis_total.plot(kind='bar', stacked=True, grid=False, legend=False, sort_columns=True)
+
+
+# In[19]:
+
+# How frequent are different mutations across different cancer types?
+sns.heatmap(mut_tis_total)
+
+
+# Visualizing the input data here is key. The heterogeneity of the mutations across tissues is apparent for this particular pathway. In comparison with `TP53` mutations, it appears that Hippo signaling impacts different tissues with much higher diversity.
+# 
+# Looking closer at the plots above, it is evident that several tissues do not demonstrate aberrations (at least at the mutation level) in Hippo signaling. Specifically, it appears that cancers with gender specificity like testicular cancer and uterine cancer  are _not_ impacted. Therefore, because of this artificial imbalance, if Cognoma were to include these cancers in the classifier, it **will** key in on gender specific signal (i.e. genes that are only on the Y chromosome, or X inactivation genes).  
+
+# In[20]:
+
+# Build final y matrix for classifier
+y = y_full.max(axis=1)
+
+# Determine which tissues to filter
+filter_tissue = pd.DataFrame([y, y_full['disease']], index=['sample', 'disease']).T
+tissue_decision = filter_tissue.groupby('disease').sum()
+
+
+# In[21]:
+
+# How many samples in each tissue that have Hippo signaling aberrations
+tissue_decision
+
+
+# ## Filter Data by Tissue
+# 
+# This is a crucial step that is different from previous classifiers
+
+# In[22]:
+
+# Technically, this is a hyper parameter, but for simplicity, set here
+filter_limit = 30
+filtered_mut_total = mut_tis_total[tissue_decision['sample'] >= filter_limit]
+
+
+# This filters out:
+# 
+# 1. Adrenocortical cancer
+# 2. Cholangiocarcinoma
+# 3. Diffuse large B-cell lymphoma
+# 4. Glioblastoma multiforme
+# 5. Kidney chromophobe
+# 6. Ovarian serous cystadenocarcinoma
+# 7. Pancreatic adenocarcinoma
+# 8. Pheochromocytoma & paraganglioma
+# 9. Prostate adenocarcinoma
+# 10. Testicular germ cell tumor
+# 11. Thymoma
+# 12. Thyroid carcinoma
+# 13. Uterine carcinosarcoma
+# 14. Uterine corpus endometrioid carcinoma
+# 15. Uveal melanoma
+
+# In[23]:
+
+# Distribution of mutation counts after filtering
+sns.heatmap(filtered_mut_total, annot=True)
+
+
+# In[24]:
+
+clinical = clinical[clinical['disease'].isin(filtered_mut_total.index)]
+clinical.disease.value_counts()
+
+
+# In[25]:
```

Comment:
More columns would be nice. In notebook, when displaying full tables, might as well use the space. So like n_pos and n_neg would be helpful.